common/src/hash.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Hash-based containers and helpers.
+
+use crate::hasher_builder::HasherBuilder;
+
+/// A HashMap that replaces the default hasher with an implementation that
+/// relies on mcrand for randomess.
+/// See [hashbrown::HashMap] and [HasherBuilder]
+pub type HashMap<K, V> = hashbrown::HashMap<K, V, HasherBuilder>;
+
+/// A HashSet that replaces the default hasher with an implementation that
+/// relies on mcrand for randomess.
+/// See [hashbrown::HashSet] and [HasherBuilder]
+pub type HashSet<K> = hashbrown::HashSet<K, HasherBuilder>;
```

Comment:
Eventually the the HasherBuilder should be re-reviewed. I believe it's using some hashing primitives which were made to be fast back in the day but there's been some evolution in the technology since then. I'll bring this us for re-evaluation.

Reply:
Sounds reasonable. Wanna file a tracking issue?

Reply:
Here we are! https://github.com/mobilecoinfoundation/mobilecoin/issues/1937